packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -216,6 +224,24 @@ class TimeInput extends React.Component {
     });
   }
 
+  /* When Time-Input is used with date-time-picker time-input In missing hour scenario Time-Input skips state update due to
+     prevProps and props comparsion. To handle this scenario propValue is being validated with prevState values in below method
+     to ensure state update happens on time value change.
+  */
+  hasTimeValueUpdated(prevState) {
+    let hour = TimeUtil.splitHour(this.props.value);
+    if (this.props.variant === TimeUtil.FORMAT_12_HOUR) {
```

Comment:
Reminder: Depending on the merge order between this and #898, `this.props.variant` here needs to be updated to `TimeUtil.getVariantFromLocale(this.props.variant)`.

Reply:
Thank you for adding remainder here. will follow-up with #898 PR and will update these changes accordingly.